agent_build_refactored/managed_packages/managed_packages_builders.py

```diff
@@ -60,70 +62,62 @@
-
-
-The 'scalyr-agent-python3' package provides Python interpreter that is specially built to be used by the agent.
-    Its main feature that it is built against the oldest possible version of gLibc, so it has to be enough to maintain
+The package provides the "embedded" Python interpreter that is specially built to be used by the agent.
+    It is built against the oldest possible version of gLibc, so it has to be enough to maintain
     only one build of the package in order to support all target systems.
 
     One of the features on the package, is that is can use system's OpenSSL if it has appropriate version, or
     fallback to the OpenSSL library which is shipped with the package. To achieve that, the Python interpreter from the
-    package contains multiple versions of it's 'OpenSSL-related' C bindings - _ssl.cpython*.so and _hashlib.cpython.so
+    package contains multiple versions of it's 'OpenSSL-related' C bindings - _ssl.cpython*.so and _hashlib.cpython.so.
     On each new installation/upgrade of the package or user's manual run of the command
-    `/opt/scalyr-agent-2-dependencies/bin/agent-python3-config initialize`, the package follows next steps in order to
+    `/opt/scalyr-agent-2/bin/agent-python3-config initialize`, the package follows next steps in order to
     resolve OpenSSL library to use:
         - 1: Make Python interpreter to use original ssl C bindings. In this case when 'ssl' or 'hashlib' module is
             imported, the originally compiled '_ssl.cpython*.so' and '_hashlib.cpython.so' bindings are used.
-            Those bindings, from the default, use system's dynamic linker in order to find and link appropriate OpenSLL
-            library if it is presented on system. First it tries to find OpenSSL version - 3+.
+            Those bindings, by the default, use system's dynamic linker in order to find and link appropriate OpenSLL
+            library if it is presented on system. First it tries to find OpenSSL version 3+.
         - 2: If first step is not successful and system does not have appropriate OpenSSL 3, then
             we replace previous  '_ssl.cpython*.so' and '_hashlib.cpython.so' C bindings with the same bindings, but
             that are compiled and linked against the OpenSSL 1.1.1+, and will look for OpenSSL 1.1.1+. Even though the
             _ssh and _hashlib modules may resolve multiple variants of the OpenSSL libraries, they are able to resolve
             them only for the specific major version they were compiled for. So we have to have multiple variants of the
             ssl C bindings that are compiled and linked against both OpenSSL 1.1.1+ and 3.0+.
 
-        -3: If OpenSSL 1.1.1+ is also not presented in a system, then we fallback to the 'embedded' OpenSSL library that
+        - 3: If OpenSSL 1.1.1+ is also not presented in a system, then we fallback to the 'embedded' OpenSSL library that
```

Comment:
So a version of OpenSSL is embedded is support older systems that don't have a more recent version installed?

One thing that's a little bit of a concern is that OpenSSL generally has a frequent release cycle because of discovered vulnerabilities & embedding it would mean the release cycle for the Scalyr agent should try to match it.
But then again the agent doesn't host services so maybe it's not much of a concern.

Reply:
> So a version of OpenSSL is embedded is support older systems that don't have a more recent version installed?

Correct.

Yeah, that's why we try to find the system's OpenSSL first (because it has be maintained by the distribution in a more timely manner) and only fallback to the embedded one when there are no other options. In this case yeah, we'd better track the OpenSSL update as frequent as possible. I also have a draft for a script that has to track version of OpenSSL and if there's a new version, it creates PR with update.